Tetris Clone C++/TetrisPiece.cpp

```diff
@@ -29,14 +30,31 @@ void TetrisPiece::moveLeft() {
 
 void TetrisPiece::moveRight() {
     if (this->offsetFree(1, 0)) {
-        this->updatePosition(this->col+1, this->row+0);
+        this->updatePosition(this->col+1, this->row);
     }
 }
 
 void TetrisPiece::moveDown() {
     if (this->offsetFree(0, 1)) {
         this->updatePosition(this->col, this->row+1);
+    } else {
+        this->lock();
+    }
+}
+
+int TetrisPiece::castDown(bool for_real) {
+    int index = 1;
+    std::cout << "right before loop in TetrisPiece::castDown" <<  std::endl;
+    assert(not this->offsetFree(0, ROWS + 20));
+    while (this->offsetFree(0, index)) {
+        index++;
     }
+    index--;
```

Comment:
add a comment explaining this, if you can refactor using a do while, or some other kind of logic, please do.

Gillett haz ideaz